src/mavedb/routers/mapped_variant.py

```diff
@@ -0,0 +1,44 @@
+from typing import Any, List, Optional
+
+from fastapi import APIRouter, Depends
+from fastapi.exceptions import HTTPException
+from sqlalchemy.orm import Session
+from sqlalchemy.exc import MultipleResultsFound
+
+from mavedb import deps
+
+from mavedb.models.mapped_variant import MappedVariant
+from mavedb.models.variant import Variant
+from mavedb.view_models import mapped_variant
+
+async def fetch_mapped_variant_by_variant_urn(db, urn: str) -> Optional[MappedVariant]:
+    """
+    Fetch one mapped variant by variant URN.
+
+    :param db: An active database session.
+    :param urn: The variant URN.
+    :return: The mapped variant, or None if the URL was not found or refers to a private score set not owned by the specified
+        user.
+    """
+    try:
+        variant = db.query(Variant).filter(Variant.urn == urn).one_or_none()
+        if not variant:
+            raise HTTPException(status_code=404, detail=f"Mapped variant with URN {urn} not found")
+        item = db.query(MappedVariant).filter(MappedVariant.variant_id == variant.id).one_or_none()
+    except MultipleResultsFound:
+        raise HTTPException(status_code=500, detail=f"Multiple variants with URN {urn} were found.")
+    if not item:
+        raise HTTPException(status_code=404, detail=f"Mapped variant with URN {urn} not found")
+    return item
+
+router = APIRouter(prefix="/api/v1/mappedVariants", tags=["mapped variants"], responses={404: {"description": "Not found"}})
+
+@router.get("/{urn}", status_code=200, response_model=mapped_variant.MappedVariant, responses={404: {}, 500: {}})
+async def show_mapped_variant(
+    *, urn: str, db: Session = Depends(deps.get_db)
+) -> Any:
+    """
+    Fetch a mapped variant by URN.
+    """
+
+    return await fetch_mapped_variant_by_variant_urn(db, urn)
```

Comment:
I think it makes more sense to just have the logic within `fetch_mapped_variant_by_variant_urn` here instead of wrapping it inside a separate function, similar to how we fetch existing experiments and scoresets via URN here: https://github.com/VariantEffect/mavedb-api/blob/ef48654cbc8fc7dd4649e8b0716ac2f60789f5f0/src/mavedb/routers/experiments.py#L75-L84.

That way we can keep these router files clean of any non-router processing functions. If you wanted to retain a separate function, I would create a new file `src/mavedb/variants.py` and put the function there. I think this is more applicable to search though. Again, from the experiments router: https://github.com/VariantEffect/mavedb-api/blob/ef48654cbc8fc7dd4649e8b0716ac2f60789f5f0/src/mavedb/routers/experiments.py#L65-L72

Reply:
I'm not sure what new tasks Samriddhi will do for Mapped Variants so that I leave it in a separating file. Fetching existing scoreset is in routers/scoreset.py. I prefer to move it to routers/scoreset.py rather than routers/experiment.py if there's only one fetching mapped variants function because mapped variants link to scoreset variants.

Reply:
Oh yeah no worries, it makes sense in `scoresets.py`. I was just giving an example of how when we do searching in `experiments.py` we have a separate helper function for the search (`_search_experiments`). I think that would be applicable here too. Instead of having two separate functions, we could just combine them into one:
```
@router.get("/{urn}", status_code=200, response_model=mapped_variant.MappedVariant, responses={404: {}, 500: {}})
async def show_mapped_variant(
    *, urn: str, db: Session = Depends(deps.get_db)
) -> Any:
    """
    Fetch a mapped variant by URN.
    """
    try:
        variant = db.query(Variant).filter(Variant.urn == urn).one_or_none()
        if not variant:
            raise HTTPException(status_code=404, detail=f"Mapped variant with URN {urn} not found")
        item = db.query(MappedVariant).filter(MappedVariant.variant_id == variant.id).one_or_none()
    except MultipleResultsFound:
        raise HTTPException(status_code=500, detail=f"Multiple variants with URN {urn} were found.")
    if not item:
        raise HTTPException(status_code=404, detail=f"Mapped variant with URN {urn} not found")
    return item
```

We likely won't need to reuse the fetch logic outside of the router, so I think it's ok to just have one function in the router file that handles all the fetch logic.

Reply:
I think it depends on whether this function will be used more than one time. If only `show_mapped_variant` needs it, it's better to combine them such as `fetch_experiment` in routers/experiment.py. Otherwise, it's good to separate them such as `fetch_scoreset_by_urn` in routers/scoreset.py. Currently MappedVariant is a draft model. We'll change this model's attributes and add more functions when Samriddhi join us.

Reply:
Agreed, makes sense to me.